js/src/api/transport/ws/ws.js

```diff
@@ -249,6 +292,38 @@ export default class Ws extends JsonRpcBase {
     });
   }
 
+  subscribe (method, callback, ...params) {
+    return new Promise((resolve, reject) => {
+      const id = this.id;
+      const json = this.encode(method, params);
+      const subscription = true;
+
+      this._messages[id] = { id, method, params, json, resolve, reject, callback, subscription };
+
+      this._send(id);
+    });
+  }
+
+  unsubscribe (messageId) {
+    return new Promise(() => {
+      const id = this.id;
+      const { subId, method } = this._messages[messageId];
+      const params = [subId];
+      const uMethod = method === 'eth_subscribe' ? 'eth_unsubscribe' : 'parity_unsubscribe';
+      const json = this.encode(uMethod, params);
+
+      this._messages[id] = { id, 'method': uMethod, params, json,
+        resolve: _ => {
+          delete this._messages[messageId];
+          method === 'eth_subscribe' ? delete this._subscriptions['eth_subscription'][subId] : delete this._subscriptions['parity_subscription'][subId];
```

Comment:
`delete this._subscriptions[method === 'eth_subscribe' ? ...][subId];`

Also I think this check is done more often than necessary, can't we store the subscription names and unsubscribe method names in `this_messages`? I think it would simplify a lot of code.